vision/labelDetection/Program.cs

```diff
@@ -0,0 +1,141 @@
+        /// <returns>a list of labels detected by the Vision API for the image.
+        /// </returns>
+        public IList<AnnotateImageResponse> DetectLabels(
+            VisionService vision, string imagePath)
+        {
+            Console.WriteLine("Detecting Labels...");
+            // Convert image to Base64 encoded for JSON ASCII text based request   
+            byte[] imageArray = System.IO.File.ReadAllBytes(imagePath);
+            string imageContent = Convert.ToBase64String(imageArray);
+            // Post label detection request to the Vision API
+            // [START construct_request]
+            var responses = vision.Images.Annotate(
+                new BatchAnnotateImagesRequest()
+                {
+                    Requests = new[] {
+                    new AnnotateImageRequest() {
+                        Features = new [] { new Feature() { Type =
+                          "LABEL_DETECTION"}},
+                        Image = new Image() { Content = imageContent }
+                    }
+               }
+                }).Execute();
+            return responses.Responses;
+            // [START construct_request]
+        }
+        // [END detect_labels]
+
+        // [START run_application]
+        private static void Main(string[] args)
+        {
+            SamplesUtil.InvokeMain(() =>
+            {
+                var sample = new LabelDetectionSample();
+                sample.MainFunction(args);
+            });
+        }
+
+        void MainFunction(string[] args)
+        {
+            LabelDetectionSample sample = new LabelDetectionSample();
+            string imagePath = null;
+            if (args.Length == 0)
+            {
+                Console.WriteLine(usage);
+                return;
+            }
+            imagePath = args[0];
+            // Create a new Cloud Vision client authorized via Application 
+            // Default Credentials
+            VisionService vision = CreateAuthorizedClient();
+            // Use the client to get label annotations for the given image
+            // [START parse_response]
+            IList<AnnotateImageResponse> result = DetectLabels(
+                vision, imagePath);
+            // Check if label annotations were found
+            if (result != null)
+            {
+                Console.WriteLine("Labels for image: " + imagePath);
+                // Loop through and output label annotations for the image
+                foreach (var response in result)
+                    foreach (var label in response.LabelAnnotations)
+                        Console.WriteLine(label.Description + " (score:"
+                        + label.Score + ")");
+ 
+            }
+            else
+            {
+                Console.WriteLine("No labels found.");
+            }
+            // [END parse_response]
+            Console.WriteLine("Press any key...");
+            Console.ReadKey();

```

Comment:
This is redundant with SamplesUtil.InvokeMain().  I removed SamplesUtil.InvokeMain() from another sample because I think it adds more distraction than it's worth.   Let's do the same here:  Keep Console.WriteLine() and Console.ReadKey(), and make this MainFunction the true Main().